### PR TITLE
Hopefully helpful updates

### PR DESCRIPTION
**Pull request by cmcfarlen**

### CODE REVIEW

File: project.clj [hunk 1]
```diff
@@ -1,7 +1,13 @@
...
-  :dependencies [[org.clojure/clojure "1.7.0"]
-                 [org.clojure/clojurescript "1.7.170"]
-                 [org.omcljs/om "1.0.0-alpha30"]
-                 [cljsjs/pouchdb "6.0.4-0"]
-                 [figwheel-sidecar "0.5.0-6" :scope "test"]])
+  :dependencies [[org.clojure/clojure "1.9.0-alpha13"]
+                 [org.clojure/clojurescript "1.9.229"]
+                 [org.omcljs/om "1.0.0-alpha47"]
+                 [cljsjs/pouchdb "6.0.4-0"]]
+
```
Comment (cmcfarlen) [inline project.clj:7]: bump to latest om, cljs and figwheel.

File: src/om_pouch/core.cljs [hunk 2]
```diff
@@ -24,14 +24,14 @@
   (let [st @state]
     (println "sssst: " st)
     (if (contains? st key)
-      {:value (into [] (map #(get-in st %)) (get st key))}
+      {:value (om/db->tree query (key st) st)}
```
Comment (cmcfarlen) [inline src/om_pouch/core.cljs:27]: idiomatic way to read from normalized state.

File: src/om_pouch/core.cljs [hunk 3]
```diff
@@ -47,7 +47,7 @@
                       (update v :item/by-id merge {id params}))))))
   :pouchput ast})
 
-(defui Row
+(defui ^:once Row
```
Comment (cmcfarlen) [inline src/om_pouch/core.cljs:50]: This prevents the def'd symbol from being redefined, but redefines all of the protocol implementations.  Reduces surprises during figwheel code reloads.

File: src/om_pouch/core.cljs [hunk 4]
```diff
@@ -118,8 +118,10 @@
           query (get-in pouchput [:params])]
       
       (let [id (:_id query)
-            nitem {:item/by-id {id {:_id id :item "I JUST CHANGED YOU"}}}]
-        (cb nitem pouchput))))
+            stable-id (random-uuid)
+            result {'pouchput {:tempids {[:item/by-id id] [:item/by-id stable-id]}}
+                    [:item/by-id stable-id] {:_id stable-id :item (str (:item query) " - I JUST CHANGED YOU")}}]
```
Comment (cmcfarlen) [inline src/om_pouch/core.cljs:123]: This is the general shape of a response from a remote that adds novelty.  Merging novelty using idents (the `[:item/by-id id]` keys) will not require a read query passed to the cb fn.

File: src/om_pouch/core.cljs [hunk 5]
```diff
@@ -118,8 +118,10 @@
...
+        (cb result))))
```
Comment (cmcfarlen) [inline src/om_pouch/core.cljs:124]: Don't pass a query for mutate responses in this case.

File: src/om_pouch/core.cljs [hunk 6]
```diff
@@ -152,7 +154,7 @@
 
 (def init-data (atom {}))
 
-(def reconciler
+(defonce reconciler
```
Comment (cmcfarlen) [inline src/om_pouch/core.cljs:157]: defonce prevents the reconciler from being redefined on figwheel reload.

File: src/user.clj [hunk 7]
```diff
@@ -0,0 +1,29 @@
+(ns user
```
Comment (cmcfarlen) [inline src/user.clj:1]: I added this namespace so I could start with `lein repl` and then just call `(start)` from there so I could eval forms in the cljs repl.

**Comment by johnomarkid** [comment]
@cmcfarlen this is incredibly helpful!! You cleared up a lot of confusion I had, and even exposed some issues I didn't realize I had. Thanks a lot!

File: src/om_pouch/core.cljs (hunk 5 above)
Comment (johnomarkid) [reply src/om_pouch/core.cljs:124]: hey @cmcfarlen this seems to merge the new item into the state, but the list does not automatically re-render the item + "I JUST CHANGED YOU" string. If I do something that calls for a re-render, the state shows with the text above. Is it working like that on your end too?

**Pull request merged**